text/0715-construct-upgrade-support.md

```diff
@@ -0,0 +1,514 @@
+# CDK Construct Upgrade Support
+
+- **Original Author(s):**: @GavinZZ
+- **Tracking Issue**: #715
+- **API Bar Raiser**: @iliapolo
+
+A new migration framework in the CDK CLI and construct library that helps developers migrate from
+older construct libraries to modern, fully supported alternatives that incorporate the latest
+features and best practices.
+
+## Working Backwards
+
+As the AWS Cloud Development Kit (CDK) evolves, we continuously deliver new features,
+performance optimizations, and reliability improvements. To maintain backward compatibility
+while enabling forward progress, the CDK team uses structured versioning strategies for critical updates.
+
+A common pattern involves releasing versioned constructs or modules (e.g., V2 variants)
+that coexist with their predecessors. For example:
+
+* The `@aws-cdk/aws-ec2-alpha` alpha module introduces the modernized `VpcV2` construct,
+designed to succeed the original Vpc in `aws-cdk-lib/aws-ec2`
+* The `TableV2` construct in `aws-cdk-lib/aws-dynamodb` replaces legacy Table implementations
+while retaining API compatibility
+
+These updates unlock enhanced capabilities. Migration between versions, however, often requires
+code changes that could trigger CloudFormation resource replacements (e.g., NAT gateways being
+recreated, DynamoDB global tables being recreated). For stateful production resources, this can
+lead to availability risks and operational complexity.
```

Comment:
These are weasel words. Be specific about the problems.

"Recreation of resources leads to data loss for stateful resources, and can lead to loss of availability for others."

I'm not sure what the "operational complexity" is referring to. What is complex about data loss?